### ISSUE TITLE
Overriding Gnome template

### ISSUE THREAD

**Issue by musjj** [danth/stylix#621]
I wanted to override a few colors for Gnome (without recreating all the needed derivations), but it looks like that the template is [hardcoded](https://github.com/danth/stylix/blob/6863412636c8f2cb3b7360f747fbd020fbfddf68/modules/gnome/colors.mustache).

There's this relevant thread: https://github.com/danth/stylix/issues/159 about moving these templates out-of-tree. Would a PR for separating the mustache template to a different repo be acceptable here?

**Comment by trueNAHO** [comment]
> I wanted to override a few colors for Gnome (without recreating all the needed derivations), but it looks like that the template is [hardcoded](https://github.com/danth/stylix/blob/6863412636c8f2cb3b7360f747fbd020fbfddf68/modules/gnome/colors.mustache).

Ideally, we would upstream (NixOS, Home Manager, or nix-darwin) relevant options to seamlessly integrate Stylix into external Nix code.

> There's this relevant thread: #159 about moving these templates out-of-tree. Would a PR for separating the mustache template to a different repo be acceptable here?

The [future Vim highlight groups](https://github.com/danth/stylix/issues/534) would allow us to get rid of mustache templates:

> the [roadmap](https://github.com/danth/stylix/issues/534) eventually causes Stylix to implement all templates without relying on external dependencies
>
> -- https://github.com/danth/stylix/issues/567#issuecomment-2403246345

**Comment by musjj** [comment]
Are the patches really suitable for upstreaming though? It contains a lot of stylix-specific changes, like removing dark mode. The style sheet itself is also fairly opinionated and custom, which is not something I usually see upstream.

**Comment by trueNAHO** [comment]
> I see, but in the case for Gnome you can't feed the values via `xdg.configFile` though. The Gnome module involves a lot of [patching](https://github.com/danth/stylix/tree/master/modules/gnome), so upstreaming them all to NixOS + home-manager seems unlikely IMO.

Yes.

> I think a more sensible short-term approach is to import the template files via the inputs in `flake.nix` (using `url = "path:./templates/..."` so you don't have to make a bunch of repositories). This will allow the user to override it. The downside is that you can't override it partially. What do you think?

This sounds like a really good short-term solution. Feel free to submit a PR.

**Comment by trueNAHO** [comment]
> > I think a more sensible short-term approach is to import the template files via the inputs in `flake.nix` (using `url = "path:./templates/..."` so you don't have to make a bunch of repositories). This will allow the user to override it. The downside is that you can't override it partially. What do you think?
>
> This sounds like a really good short-term solution. Feel free to submit a PR.

Actually, would it not be better to do this with an option, like `stylix.targets.gnome.theme`?

**Comment by musjj** [comment]
Yeah, a `stylix.targets.gnome.theme` option might be better here if that's allowed. I'm starting to think that overriding templates might not be as useful in Gnome 47 due to: https://github.com/danth/stylix/issues/618, which *may* require search-replace patching over the css files.
